pynumaflow/function/asynciter.py

```diff
@@ -0,0 +1,23 @@
+import asyncio
+
+from pynumaflow._constants import STREAM_EOF
+
+
+class NonBlockingIterator:
+    """An Async Interator backed by a queue"""
+
+    __slots__ = "_queue"
```

Comment:
```suggestion
    __slots__ = ("_queue",)
```
__slots__ takes in a tuple usually.v

Reply:
i started with a tuple, but the editor/lint wont let me keep the parenthesis.